src/functions/helpers.ts

```diff
@@ -710,3 +710,25 @@ function toCellValue(value: any): CellValue {
 export function isMatrixArgValue(x: ArgValue): x is MatrixArgValue {
   return Array.isArray(x) && Array.isArray(x[0]);
 }
+
+/**
+ * Flatten an array of items, where each item can be a single value or a 2D array, and apply the
+ * callback to each element.
+ *
+ * The 2D array are flattened row first.
+ */
+export function flattenRowFirst<T, M>(items: Array<T | Matrix<T>>, callback: (val: T) => M): M[] {
```

Comment:
Nope this doesn't work, since the callback is typed `(val: T) => M`. I could do `(val) => val as M` I guess but I'd rather not :p 
This is never used without a callback ATM anyway.